Library/Homebrew/config.rb

```diff
@@ -42,7 +42,8 @@
   (OS.mac? ? "~/Library/Logs/Homebrew/" : "~/.cache/Homebrew/Logs")).expand_path
 
 # Must use /tmp instead of $TMPDIR because long paths break Unix domain sockets
-HOMEBREW_TEMP = Pathname.new(ENV.fetch("HOMEBREW_TEMP", "/tmp"))
+# IF ANDROID
+HOMEBREW_TEMP = Pathname.new(ENV.fetch("HOMEBREW_TEMP", "/data/data/com.termux/files/usr/tmp"))
 
```

Comment:
Setting `HOMEBREW_TEMP` can go in https://github.com/Linuxbrew/brew/wiki/Android

Reply:
I wanted to just use $TMPDIR, but the line above dissuaded me.
The strategy I've used in past for such a problem is to create a short symlink (e.g. /tmp/t) that points to the arbitrary-length TMPDIR, then use the shorter path as TMPDIR. Of course, if /tmp isn't writable, it seems best to use TMPDIR as is. 

Do you think the above suggestion could fit upstream?

Reply:
My guess is that they'd prefer to keep the existing behaviour, using `HOMEBREW_TEMP` and ignoring `TMPDIR`. An Android specific installer for Linuxbrew could set the environment variables needed for Android.

Reply:
Done: https://github.com/Linuxbrew/brew/wiki/Android